### ISSUE TITLE
Error when running python scripts/main.py

### ISSUE THREAD

**Issue by Mister-Carter** [Significant-Gravitas/Auto-GPT#498]
### Duplicates

- [X] I have searched the existing issues

### Steps to reproduce 🕹

Follow all steps in setup tutorial.

### Current behavior 😯

I get the error below: Traceback (most recent call last):
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\connection.py", line 174, in _new_conn
    conn = connection.create_connection(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\util\connection.py", line 72, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno 11001] getaddrinfo failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
                       ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\connectionpool.py", line 386, in _make_request
    self._validate_conn(conn)
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\connectionpool.py", line 1042, in _validate_conn
    conn.connect()
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\connection.py", line 363, in connect
    self.sock = conn = self._new_conn()
                       ^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x00000299CE7CA4D0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Mister Carter\Auto-GPT\scripts\main.py", line 288, in <module>
    memory = PineconeMemory()
             ^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\Auto-GPT\scripts\config.py", line 17, in __call__
    cls._instances[cls] = super(
                          ^^^^^^
  File "C:\Users\Mister Carter\Auto-GPT\scripts\memory.py", line 30, in __init__
    if table_name not in pinecone.list_indexes():
                         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\pinecone\manage.py", line 185, in list_indexes
    response = api_instance.list_indexes()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\pinecone\core\client\api_client.py", line 776, in __call__
    return self.callable(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\pinecone\core\client\api\index_operations_api.py", line 1132, in __list_indexes
    return self.call_with_http_info(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\pinecone\core\client\api_client.py", line 838, in call_with_http_info
    return self.api_client.call_api(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\pinecone\core\client\api_client.py", line 413, in call_api
    return self.__call_api(resource_path, method,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\pinecone\core\client\api_client.py", line 200, in __call_api
    response_data = self.request(
                    ^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\pinecone\core\client\api_client.py", line 439, in request
    return self.rest_client.GET(url,
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\pinecone\core\client\rest.py", line 236, in GET
    return self.request("GET", url,
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\pinecone\core\client\rest.py", line 202, in request
    r = self.pool_manager.request(method, url,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\request.py", line 74, in request
    return self.request_encode_url(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\request.py", line 96, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\poolmanager.py", line 376, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\connectionpool.py", line 815, in urlopen
    return self.urlopen(
           ^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\connectionpool.py", line 815, in urlopen
    return self.urlopen(
           ^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\connectionpool.py", line 815, in urlopen
    return self.urlopen(
           ^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\connectionpool.py", line 787, in urlopen
    retries = retries.increment(
              ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mister Carter\AppData\Local\Programs\Python\Python311\Lib\site-packages\urllib3\util\retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='controller.your-pinecone-region.pinecone.io', port=443): Max retries exceeded with url: /databases (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x00000299CE7CA4D0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))

### Expected behavior 🤔

The program should operate as described in the docs.

### Your prompt 📝

```yaml
# Name:  Algorithmic Trading GPT
Role:  An AI that creates crypto currency trading algorithm ideas, comes up with parameters to optimize each idea, optimizes each parameter for expectancy using out of sample data from walk forward analysis, analyzes the out of sample results for the expectancy metric and comes up with new ways to improve expectancy for each trading strategy, does another walk forward analysis for each strategy and picks the best parameters for each strategy, uses walk forward analysis tests to determine which combinations of strategies works best to optimize for out of sample expectancy, uses the combined data of all the out of sample expectancy metrics from the walk forward anlaysis testing to determine which strategy or combination of strategies will provide the best expectancy, then deployes that strategy on a decentralized crypto currency exchange that offers futures contract trading, and finally provides the user with instructions for how to use the finalized crypto currency trading algorithm on the decentralized crypto currency futures trading platform.
Goals: ['come up with 10 different bitcoin trading strategies that includes many different parameters that can be optimized for each bitcoin trading strategy.', 'Optimize each parameter for each strategy for the expectancy metric using out of sample data derived from a walk forward analysis performed on each strategy.', 'Analyze the results of the expectancy metric for the out of sample data in walk forward analysis for each strategy to determine which strategies performed the best. ', 'Combine the best performing strategies into a single strategy that includes the all of the best performing strategies and optimize the expectancy metric for the out of sample data in a new walk forward analysis for the new single strategy.', 'deploy this strategy on a decentralized bitcoin futures trading platform and provide instructions to the user that teaches the user how the algorithmic trading strategy works and how to utilize it.']
```

**Comment by richbeales** [comment]
duplicate of various pinecone issues https://github.com/Torantulino/Auto-GPT/issues/462 https://github.com/Torantulino/Auto-GPT/issues/490 make sure your pinecone environment variables are set.

**Comment by barbsblc** [comment]
I put the error message on Chat GPT4 and asked it to translate the problem in plain English and here is what it said. I hope it helps anyone. 

"In plain English, your program is trying to connect to a server, but it's failing to find the server's address, which is causing the error.

The problem is likely related to one of the following:

- Internet connection issue: Make sure you're connected to the internet, and your connection is stable. If you're using a proxy, check if it's configured correctly.
- DNS issue: The server's address might not be resolved correctly. Try restarting your computer, resetting your router, or using an alternative DNS server, such as Google's public DNS (8.8.8.8).
- Server's address: Make sure the server's address is correct in your code or configuration. It should be something like "controller.your-pinecone-region.pinecone.io" but with the proper Pinecone region.

To fix the issue, try the following steps:

- Check your internet connection and make sure it's working correctly.
- Verify the server's address in your code or configuration. If it's incorrect or missing the proper Pinecone region, update it.
- Restart your computer and router.
- If the issue persists, try using an alternative DNS server, like Google's public DNS (8.8.8.8).

Once you've addressed the problem, run the code again, and it should work as expected."

I'm still going to test these ideas. The internet connection here is indeed bad today.

**Comment by Explorergt92** [comment]
@Mister-Carter is this still an issue for you or can this be closed?

**Comment by Pwuts** [comment]
Closing as duplicate. Feel free to reopen if not.
